wordpress-mobile/WordPress-Android
# Adds: Logic for navigating to Static pages

Parents #18109 , #18110 , #18111 

## Description
### Navigating to Static Posters
This PR covers the core parts for managing Jetpack Static Posters phase:
- Show the Static Page when:
-- Tapping on the Reader tab
-- Tapping on the Notification tab
-- Tapping on the Stats menu item
-- Wen navigating into the app from a Stats shortcut, widget, or deep link
-- When navigating into the app from a Notification shortcut or widget

### Disables the ability to Add Stories in Static posters phase

## **TestSetup**
- Remove all other versions of WP & JP from your device
- Install WP from this PR
- Set up WP to handle deep links
    - Navigate to the device app settings for WP > Open by default
    - Tap on the Add link > Select all 3 supported links
    - Click Add
- Login to the app
- Navigate to Me > App Settings > Privacy Settings  and enable Collect Information
- Do not enable/disable the `jp_removal_static_posters`, it should be off after fresh install
- Locate the app icon on your device and long press on it
- Add a stats widget to your homescreen

 
**Test - Non Static Poster state  is not enabled**

- Tap on MySite tab
- ✅ Verify the dashboard is visible 
- Tap on the reader and notification tabs
- ✅ Verify the reader and notification content are shown
- Tap on the FAB
- ✅ Verify Story Post option is visible
- Tap on the “Stats” feature (either through the menu item, stats card, or quick links)
- ✅ Verify you are taken to the Stats area
- Navigate back to My Site
- Tap on Posts
- Tap on FAB
- ✅ Verify Story Post option is visible
- Background the app 
- Locate the stats widget you created during setup and tap on it
- ✅ Verify the app is launched and you are taken to the Stats area
- Background the app 
- Locate the app icon and long press on it
- Tap on the stats shortcut
- ✅ Verify the app is launched and you are taken to the Stats area
- Background the app 
- Locate the app icon and long press on it
- Tap on the notifications shortcut
- ✅ Verify the app is launched and you are taken to the Notifications tab showing notification content
- Deep link into the app using the following 
`<p><a href="intent://stats/#Intent;scheme=wordpress;package=org.wordpress.android.prealpha;end;">WordPress Stats</a></p>`
- ✅ Verify the app is launched and you are taken to the Stats area

**Test - Static Poster state is enabled**

- Navigate to Me > App Settings > Debug Settings
- Enable `jp_removal_static_posters`
- Navigate back to My Site Tab
- ✅ Verify the dashboard is not visible 
- Tap on the reader and notification tabs
- ✅ Verify the reader and notification content are not shown, but rather the static page is shown
- Tap on the FAB
- ✅ Verify Story Post option is not visible
- Tap on the “Stats” menu item 
- ✅ Verify you are shown the static page
- Navigate back to My Site
- Tap on Posts
- Tap on FAB
- ✅ Verify Story Post option is not visible (You should be taken straight into the editor)
- Background the app 
- Locate the stats widget you created during setup and tap on it
- ✅ Verify the app is launched and you are shown the static page
- Background the app 
- Locate the app icon and long press on it
- Tap on the stats shortcut
- ✅ Verify the app is launched and you are shown the static page
- Background the app 
- Locate the app icon and long press on it
- Tap on the notifications shortcut
- ✅ Verify the app is launched and you are taken to the Notification tab showing the static page
- Deep link into the app using the following:
``<p><a href="intent://stats/#Intent;scheme=wordpress;package=org.wordpress.android.prealpha;end;">WordPress Stats</a></p>``
- ✅ Verify the app is launched and you are taken to the static page

## Regression Notes
1. Potential unintended areas of impact
The tabs don't work as expected, the stats view shows when it shouldn't, there are overlapping fragments in the main view.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and individual unit tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
